### ISSUE TITLE
Document start point methods properly

### ISSUE THREAD

**Issue by jvanhoefer** [ICB-DCM/pyPESTO#709]
**Bug description**
[A clear and concise description of what the bug is.]

**Expected behavior**
[A clear and concise description of what you expected to happen.]

**To reproduce**
[A minimal example to reproduce this error, ideally a simple python script.]

**Environment**
- Operating system: [e.g. Ubuntu 20.04, macOS 11.2.1]
- `pypesto` version: [via `pip show pypesto`]
- Python version: [e.g. `python3 -V`]

**Comment by yannikschaelte** [comment]
agreed, potentially one may also like to refactor by derivation from an abstract base class, to document expected arguments properly.
